internal/resources/machinelearning/resource_job_test.go

```diff
@@ -18,53 +18,61 @@ func TestAccResourceJob(t *testing.T) {
 	testutils.CheckCloudInstanceTestsEnabled(t)
 
 	randomName := acctest.RandomWithPrefix("Test Job")
+	defaultIDRegexp := regexp.MustCompile(`^[a-zA-Z0-9-_]+$`)
 
 	var job mlapi.Job
 	resource.ParallelTest(t, resource.TestCase{
 		ProtoV5ProviderFactories: testutils.ProtoV5ProviderFactories,
-		CheckDestroy:             testAccMLJobCheckDestroy(&job),
+		CheckDestroy: resource.ComposeTestCheckFunc(
+			testAccMLJobCheckDestroy(&job),
+			testAccDatasourceCheckDestroy(),
+		),
 		Steps: []resource.TestStep{
 			{
-				Config: testutils.TestAccExampleWithReplace(t, "resources/grafana_machine_learning_job/job.tf", map[string]string{
+				// Note for the reader: these tests construct a datasource & a job, where the Job's datasource id is set by terraform.
+				Config: machinelearningDatasource + testutils.TestAccExampleWithReplace(t, "resources/grafana_machine_learning_job/job.tf", map[string]string{
 					"Test Job": randomName,
+					"= 10":     "= split(\":\", grafana_data_source.foo.id)[1]", // as it of form ^(0|1):[a-zA-Z0-9-_]+$, we only use later part
```

Comment:
It'd be good if ML supported uids as IDs are being deprecated but that's for another Pr